File: Python Pattern Programs/Wave Patterns/Pattern 8.py
wH = 4  # 1
wL = 4  # 2
pc = wH - 1  # 3

for x in range(0, wH + 1):
    for y in range(0, (wH * wL * 2)):
        if (y % (wH * 2) == pc):
            print("/", end="")
        elif (y % (wH * 2) == wH + x):
            print("\\", end="")
        else:
            print(" ", end="")  # single space
    pc -= 1
    print()

    """
1) change value to increase/decrease the height of the wave
2) change value to increase/decrease the length of the wave
3) cond for printing
    """

File: C# Pattern Programs/Wave Patterns/Pattern 8.cs
using System;

 class Wave_8
 {
  public static void Main(string[] args)
  {
   int wH = 4; //wave height-> change value to increase or decrease the height of wave


   int wL = 4; //wave length->change value to increase or decrease the length of wave


   int x = wH - 1; //if cond for printing


   for (int i = 0; i <= wH; i++)
   {
    for (int j = 0; j < wH * wL * 2; j++)
    {

     if (j % (wH * 2) == x)
     {
      Console.Write("/");
     }
     else if (j % (wH * 2) == wH + i)
     {
      Console.Write("\\");
     }
     else
     {
      Console.Write(" ");
     }
    }
    x--;
    Console.WriteLine();
   }

  Console.ReadKey(true);
  }
 }
